libs/sdk-embedding/src/iframe/kd.ts

```diff
@@ -1023,4 +1039,131 @@ export namespace EmbeddedKpiDashboard {
+    export interface ISetFilterParentsAttributeFilter {
+        attributeFilter: {
+            displayForm: ObjRef;
+        };
+    }
+
+    /**
+     * @public
+     */
+    export type SetFilterParentsItemFilter = ISetFilterParentsAttributeFilter;
+
+    /**
+     * @public
+     */
+    export interface ISetFilterParentsItemParent {
+        parent: SetFilterParentsItemFilter;
+        connectingAttribute: ObjRef;
+    }
+
+    /**
+     * @public
+     */
+    export interface ISetFilterParentsItem {
+        filter: SetFilterParentsItemFilter;
+        parents: ISetFilterParentsItemParent[];
+    }
+
+    /**
+     * @public
+     */
+    export interface ISetFilterParentsDataBody {
+        filters: ISetFilterParentsItem[];
+    }
+
+    /**
+     * @public
+     */
+    export type SetFilterParentsCommandData = IGdcKdMessageEnvelope<
+        GdcKdCommandType.SetFilterParents,
+        ISetFilterParentsDataBody
+    >;
+
+    /**
+     * @public
+     */
+    export function isSetFilterParentsCommandData(obj: unknown): obj is SetFilterParentsCommandData {
+        return getEventType(obj) === GdcKdCommandType.SetFilterParents;
+    }
+
+    /**
+     * @public
+     */
+    export type SetFilterParentsFinishedData = IGdcKdMessageEnvelope<
+        GdcKdEventType.SetFilterParentsFinished,
+        IKdAvailableCommands
+    >;
+
+    /**
+     * # SetFilterParentsErrorCode
+     * @public
+     * Error type within AppCommandFailed event body when setFilterParents command is not successfull
+     *
+     * ## Example
+     * ```
+     * {
+     *   errorCode: "invalidFilterDisplayForm"
+     * }
+     * ```
+     */
+    export enum SetFilterParentsErrorCode {
+        /**
+         * Command data format is invalid
+         */
+        InvalidDataFormat = "invalidDataFormat",
+
+        /**
+         * Attribute filter display form has invalid ref.
+         */
+        InvalidAttributeFilterDisplayForm = "invalidAttributeFilterDisplayForm",
+
+        /**
+         * Parent filter display form has invalid ref or does not exist.
+         */
+        InvalidParentFilterDisplayForm = "invalidParentFilterDisplayForm",
+
+        /**
+         * Filter is not in dashboard.
+         */
+        FilterNotFound = "filterNotFound",
+
+        /**
+         * Filter can not depend on itself.
+         */
+        CircularDependency = "circularDependency",
+
+        /**
+         * Connecting attribute is invalid.
+         */
+        InvalidConnectingAttribute = "invalidConnectingAttribute",
+
+        /**
+         * Connecting attribute is not shared between filter and parent.
+         */
+        IncompatibleConnectingAttribute = "incompatibleConnectingAttribute",
+
+        /**
+         * Multiple filters with same id in single command
+         */
+        DuplicateFilters = "duplicateFilters",
+
+        /**
+         * Multiple parents with same id in single filter
+         */
+        DuplicateParents = "duplicateParents",
+    }
```

Comment:
Error codes for other messages are in AD/KD for some reason so they are not accessible to SDK users. This is much better approach IMO